src/kofta_metadata.erl

```
@@ -165,7 +167,7 @@ disconnected(_Msg, _From, State) ->
             {disconnected, NewTimeout}
     end,
 
-    NewState = State#st{last_reconnect=os:timestamp()},
+    NewState = State#st{last_reconnect=os:timestamp(), clients=Clients1},
     {reply, {error, broker_down}, NextState, NewState, Timeout}.
```

Bah, I misread the return and thought it was silently dropping the message. I'm +1 on the `{error, broker_down}` reply.